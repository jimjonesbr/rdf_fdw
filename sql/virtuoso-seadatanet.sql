
CREATE SERVER seadatanet
FOREIGN DATA WRAPPER rdf_fdw
OPTIONS (
  endpoint 'https://edmo.seadatanet.org/sparql/sparql',
  format 'xml',
  enable_pushdown 'true'
);

CREATE FOREIGN TABLE seadatanet (
  identifier text OPTIONS (variable '?id'),
  name  varchar OPTIONS (variable '?name'),
  modified timestamp OPTIONS (variable '?modified'),
  notation int OPTIONS (variable '?notation')
) SERVER seadatanet OPTIONS 
  (log_sparql 'true',
   sparql '
    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>PREFIX foaf: <http://xmlns.com/foaf/0.1/>
    PREFIX ns: <http://www.w3.org/2006/vcard/ns#>PREFIX dct: <http://purl.org/dc/terms/>
    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>SELECT* WHERE{?id ns:hasGeo ?geo;
        <http://www.w3.org/ns/org#name> ?name;
        dct:modified ?modified ;
        skos:notation ?notation}
  ');

SELECT name, modified
FROM seadatanet
WHERE 
  modified > '2021-04-01' AND
  modified < '2021-04-30'
ORDER BY modified
LIMIT 10;

SELECT name, modified
FROM seadatanet
WHERE 
  modified BETWEEN '2021-04-01'::timestamp AND '2021-04-30'::timestamp
ORDER BY modified
LIMIT 10;

/* ################### SeaDataNet EDMO Code Country ################### */

CREATE FOREIGN TABLE edmo_country (
  code int OPTIONS (variable '?EDMO_CODE'),
  org_name  varchar OPTIONS (variable '?ORG_NAME'),
  country text OPTIONS (variable '?COUNTRY'),
  expr_col1 text OPTIONS (variable '?foo', expression '(CONCAT(UCASE(?COUNTRY), " - ", UCASE(?ORG_NAME)) AS ?foo)'),
  expr_col2 text OPTIONS (variable '?bar', expression '(STRLEN(?COUNTRY) + STRLEN(?ORG_NAME) AS ?bar)')
) SERVER seadatanet OPTIONS 
  (log_sparql 'true',
   sparql '
    PREFIX skos:<http://www.w3.org/2004/02/skos/core#>
    PREFIX rdf:<http://www.w3.org/1999/02/22-rdf-syntax-ns#>
    SELECT *
    WHERE {
	    ?EDMO_URL rdf:type <http://www.w3.org/ns/org#Organization>.
  	  ?EDMO_URL <http://www.w3.org/ns/org#name> ?ORG_NAME.
	    ?EDMO_URL skos:notation ?EDMO_CODE.
	    ?EDMO_URL <http://www.w3.org/2006/vcard/ns#country-name> ?COUNTRY.
    }
  ');

  SELECT * FROM edmo_country 
  FETCH FIRST 15 ROWS ONLY; 