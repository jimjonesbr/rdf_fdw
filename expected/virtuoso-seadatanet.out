CREATE SERVER seadatanet
FOREIGN DATA WRAPPER rdf_fdw
OPTIONS (
  endpoint 'https://edmo.seadatanet.org/sparql/sparql',
  format 'xml',
  enable_pushdown 'true'
);
CREATE FOREIGN TABLE seadatanet (
  identifier text OPTIONS (variable '?id'),
  name  varchar OPTIONS (variable '?name'),
  modified timestamp OPTIONS (variable '?modified'),
  notation int OPTIONS (variable '?notation')
) SERVER seadatanet OPTIONS 
  (log_sparql 'true',
   sparql '
    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>PREFIX foaf: <http://xmlns.com/foaf/0.1/>
    PREFIX ns: <http://www.w3.org/2006/vcard/ns#>PREFIX dct: <http://purl.org/dc/terms/>
    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>SELECT* WHERE{?id ns:hasGeo ?geo;
        <http://www.w3.org/ns/org#name> ?name;
        dct:modified ?modified ;
        skos:notation ?notation}
  ');
SELECT name, modified
FROM seadatanet
WHERE 
  modified > '2021-04-01' AND
  modified < '2021-04-30'
ORDER BY modified
LIMIT 10;
NOTICE:  SPARQL query sent to 'https://edmo.seadatanet.org/sparql/sparql':


    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>PREFIX foaf: <http://xmlns.com/foaf/0.1/>
    PREFIX ns: <http://www.w3.org/2006/vcard/ns#>PREFIX dct: <http://purl.org/dc/terms/>
    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
SELECT ?name ?modified 
 WHERE{?id ns:hasGeo ?geo;
        <http://www.w3.org/ns/org#name> ?name;
        dct:modified ?modified ;
        skos:notation ?notation FILTER(xsd:dateTime(?modified) > xsd:dateTime("2021-04-01T00:00:00.000000"))
 FILTER(xsd:dateTime(?modified) < xsd:dateTime("2021-04-30T00:00:00.000000"))
}
ORDER BY  ASC (?modified)
LIMIT 10

                                     name                                      |           modified           
-------------------------------------------------------------------------------+------------------------------
 Research Group Estuarine and Coastal Areas                                    | Tue Apr 06 14:05:11.363 2021
 Research Group Estuarine and Coastal Areas, Picardie                          | Tue Apr 06 14:07:00.26 2021
 Sorbonne University, Faculty Earth, Environment, Biodiversity                 | Wed Apr 07 15:43:33.797 2021
 Research Institute for Nature and Forest                                      | Mon Apr 12 11:54:11.073 2021
 Hellenic Centre for Marine Research, Institute of Oceanography                | Tue Apr 13 08:36:58.75 2021
 Ifremer, Finistère-North Brittany Environment and Resources Laboratory        | Tue Apr 13 15:40:57.287 2021
 Ifremer, Environment Resources Laboratory Morbihan Pays De La Loire (Lorient) | Wed Apr 14 14:18:14.387 2021
 Wageningen Environmental Research                                             | Mon Apr 19 15:51:39.603 2021
 Portuguese Institute for Sea and Atmosphere                                   | Tue Apr 20 07:50:50.827 2021
 Helmholtz-Zentrum Geesthacht, Centre for Materials and Coastal Research       | Thu Apr 22 08:17:32.397 2021
(10 rows)

SELECT name, modified
FROM seadatanet
WHERE 
  modified BETWEEN '2021-04-01'::timestamp AND '2021-04-30'::timestamp
ORDER BY modified
LIMIT 10;
NOTICE:  SPARQL query sent to 'https://edmo.seadatanet.org/sparql/sparql':


    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>PREFIX foaf: <http://xmlns.com/foaf/0.1/>
    PREFIX ns: <http://www.w3.org/2006/vcard/ns#>PREFIX dct: <http://purl.org/dc/terms/>
    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
SELECT ?name ?modified 
 WHERE{?id ns:hasGeo ?geo;
        <http://www.w3.org/ns/org#name> ?name;
        dct:modified ?modified ;
        skos:notation ?notation FILTER(xsd:dateTime(?modified) >= xsd:dateTime("2021-04-01T00:00:00.000000"))
 FILTER(xsd:dateTime(?modified) <= xsd:dateTime("2021-04-30T00:00:00.000000"))
}
ORDER BY  ASC (?modified)
LIMIT 10

                                     name                                      |           modified           
-------------------------------------------------------------------------------+------------------------------
 Research Group Estuarine and Coastal Areas                                    | Tue Apr 06 14:05:11.363 2021
 Research Group Estuarine and Coastal Areas, Picardie                          | Tue Apr 06 14:07:00.26 2021
 Sorbonne University, Faculty Earth, Environment, Biodiversity                 | Wed Apr 07 15:43:33.797 2021
 Research Institute for Nature and Forest                                      | Mon Apr 12 11:54:11.073 2021
 Hellenic Centre for Marine Research, Institute of Oceanography                | Tue Apr 13 08:36:58.75 2021
 Ifremer, Finistère-North Brittany Environment and Resources Laboratory        | Tue Apr 13 15:40:57.287 2021
 Ifremer, Environment Resources Laboratory Morbihan Pays De La Loire (Lorient) | Wed Apr 14 14:18:14.387 2021
 Wageningen Environmental Research                                             | Mon Apr 19 15:51:39.603 2021
 Portuguese Institute for Sea and Atmosphere                                   | Tue Apr 20 07:50:50.827 2021
 Helmholtz-Zentrum Geesthacht, Centre for Materials and Coastal Research       | Thu Apr 22 08:17:32.397 2021
(10 rows)

/* ################### SeaDataNet EDMO Code Country ################### */
CREATE FOREIGN TABLE edmo_country (
  code int OPTIONS (variable '?EDMO_CODE'),
  org_name  varchar OPTIONS (variable '?ORG_NAME'),
  country text OPTIONS (variable '?COUNTRY'),
  expr_col1 text OPTIONS (variable '?foo', expression '(CONCAT(UCASE(?COUNTRY), " - ", UCASE(?ORG_NAME)) AS ?foo)'),
  expr_col2 text OPTIONS (variable '?bar', expression '(STRLEN(?COUNTRY) + STRLEN(?ORG_NAME) AS ?bar)')
) SERVER seadatanet OPTIONS 
  (log_sparql 'true',
   sparql '
    PREFIX skos:<http://www.w3.org/2004/02/skos/core#>
    PREFIX rdf:<http://www.w3.org/1999/02/22-rdf-syntax-ns#>
    SELECT *
    WHERE {
	    ?EDMO_URL rdf:type <http://www.w3.org/ns/org#Organization>.
  	  ?EDMO_URL <http://www.w3.org/ns/org#name> ?ORG_NAME.
	    ?EDMO_URL skos:notation ?EDMO_CODE.
	    ?EDMO_URL <http://www.w3.org/2006/vcard/ns#country-name> ?COUNTRY.
    }
  ');
  SELECT * FROM edmo_country 
  FETCH FIRST 15 ROWS ONLY; 
NOTICE:  SPARQL query sent to 'https://edmo.seadatanet.org/sparql/sparql':


    PREFIX skos:<http://www.w3.org/2004/02/skos/core#>
    PREFIX rdf:<http://www.w3.org/1999/02/22-rdf-syntax-ns#>
    
SELECT ?EDMO_CODE ?ORG_NAME ?COUNTRY (CONCAT(UCASE(?COUNTRY), " - ", UCASE(?ORG_NAME)) AS ?foo) (STRLEN(?COUNTRY) + STRLEN(?ORG_NAME) AS ?bar) 
 WHERE {
	    ?EDMO_URL rdf:type <http://www.w3.org/ns/org#Organization>.
  	  ?EDMO_URL <http://www.w3.org/ns/org#name> ?ORG_NAME.
	    ?EDMO_URL skos:notation ?EDMO_CODE.
	    ?EDMO_URL <http://www.w3.org/2006/vcard/ns#country-name> ?COUNTRY.
    }
LIMIT 15

 code |                                              org_name                                              |  country   |                                                    expr_col1                                                    | expr_col2 
------+----------------------------------------------------------------------------------------------------+------------+-----------------------------------------------------------------------------------------------------------------+-----------
  674 | Centre of the Caspian Sea Problems, Institute of Geography                                         | Azerbaijan | AZERBAIJAN - CENTRE OF THE CASPIAN SEA PROBLEMS, INSTITUTE OF GEOGRAPHY                                         | 68
  675 | Sumgayit Center for Environmental Rehabilitation                                                   | Azerbaijan | AZERBAIJAN - SUMGAYIT CENTER FOR ENVIRONMENTAL REHABILITATION                                                   | 59
  866 | Azerbaijan Hydrometeorological Service                                                             | Azerbaijan | AZERBAIJAN - AZERBAIJAN HYDROMETEOROLOGICAL SERVICE                                                             | 48
 1760 | AzEcoLab                                                                                           | Azerbaijan | AZERBAIJAN - AZECOLAB                                                                                           | 18
 1765 | The Kulevi Oil Terminal 2000 Ltd.                                                                  | Azerbaijan | AZERBAIJAN - THE KULEVI OIL TERMINAL 2000 LTD.                                                                  | 44
 1855 | State Oil Company of Azerbaijan Republic                                                           | Azerbaijan | AZERBAIJAN - STATE OIL COMPANY OF AZERBAIJAN REPUBLIC                                                           | 51
 2437 | Institute of Zoology of the Azerbaijan, National Academy of  Sciences                              | Azerbaijan | AZERBAIJAN - INSTITUTE OF ZOOLOGY OF THE AZERBAIJAN, NATIONAL ACADEMY OF  SCIENCES                              | 79
 2438 | Institute of Geology of the Azerbaijan, National Academy of  Sciences                              | Azerbaijan | AZERBAIJAN - INSTITUTE OF GEOLOGY OF THE AZERBAIJAN, NATIONAL ACADEMY OF  SCIENCES                              | 79
 2441 | Department of Monitoring of an Environment of Caspian sea, Ministry of Ecology and Nature Resource | Azerbaijan | AZERBAIJAN - DEPARTMENT OF MONITORING OF AN ENVIRONMENT OF CASPIAN SEA, MINISTRY OF ECOLOGY AND NATURE RESOURCE | 108
 2442 | Baku State University                                                                              | Azerbaijan | AZERBAIJAN - BAKU STATE UNIVERSITY                                                                              | 31
 2443 | Institute of Ecology, Azerbaijan National Space Agency                                             | Azerbaijan | AZERBAIJAN - INSTITUTE OF ECOLOGY, AZERBAIJAN NATIONAL SPACE AGENCY                                             | 64
 2474 | Geographical Society of Azerbaijan                                                                 | Azerbaijan | AZERBAIJAN - GEOGRAPHICAL SOCIETY OF AZERBAIJAN                                                                 | 44
 2481 | Institute of Physiology of the Azerbaijan National Academy of  Sciences                            | Azerbaijan | AZERBAIJAN - INSTITUTE OF PHYSIOLOGY OF THE AZERBAIJAN NATIONAL ACADEMY OF  SCIENCES                            | 81
 2482 | Institute of Microbiology of the Azerbaijan National Academy of  Sciences                          | Azerbaijan | AZERBAIJAN - INSTITUTE OF MICROBIOLOGY OF THE AZERBAIJAN NATIONAL ACADEMY OF  SCIENCES                          | 83
 2483 | IQLIM LTD, Research and Consulting Company                                                         | Azerbaijan | AZERBAIJAN - IQLIM LTD, RESEARCH AND CONSULTING COMPANY                                                         | 52
(15 rows)

